src/core/src/main/java/org/locationtech/geogig/storage/NodePathStorageOrder.java

```diff
@@ -71,29 +109,118 @@ public int compare(String p1, String p2) {
+     * 
+     * @return {@code 512} for depth indexes {@code 0} to {@code 2}, and {@code 256} for depth
+     *         indexes bigger than {@code 2}
+     * @since 1.0
+     */
+    public static int normalizedSizeLimit(final int depthIndex) {
+        Preconditions.checkArgument(depthIndex > -1,
+                "depthIndex must be a positive integer or zero");
+        switch (depthIndex) {
+        case 0:
+        case 1:
+        case 2:
+            return 512;
+        default:
+            return 256;
+        }
+    }
+
+    /**
+     * Returns a positive integer defining in how many <i>bucket trees</i> a <i>leaf tree</i> that
+     * has reached its {@link #getNormalizedSizeLimit maximum capacity} is split into, at the given
+     * depth index.
+     * <p>
+     * E.g., if a feature tree is to be split into buckets, then this method must be called with
+     * {@code depthIndex == 0}; a bucket tree that's a direct child of a feature tree with
+     * {@code depthIndex == 1}, and so forth.
+     * <p>
+     * The combination of {@code maxBucketsForLevel(int depthIndex)} and
+     * {@link #normalizedSizeLimit(int) normalizedSizeLimit(int depthIndex)} defines the maximum
+     * capacity of a tree of a given depth.
+     * 
+     * @param depthIndex the depth index (starting at zero) of a leaf tree that needs to be split
+     *        into buckets.
+     * @return {@code 32} for depth indexes {@code 0} to {@code 2}; {@code 8} for depth indexes
+     *         {@code 3} and {@code 4}; {@code 4} for depth indexes {@code 5} and {@code 6}; and
+     *         {@code 2} for any depth index bigger than 6.
+     * @since 1.0
+     */
+    public static int maxBucketsForLevel(final int depthIndex) {
+        Preconditions.checkArgument(depthIndex > -1,
+                "depthIndex must be a positive integer or zero");
+        switch (depthIndex) {
+        case 0:
+        case 1:
+        case 2:
+            return 32;
+        case 3:
+        case 4:
+            return 8;
+        case 5:
+        case 6:
+            return 4;
+        default:
+            return 2;
+        }
+    }
+
+    /**
      * Computes the bucket index that corresponds to the given node name at the given depth.
      * 
-     * @return and Integer between zero and {@link RevTree#MAX_BUCKETS} minus one
+     * @return and Integer between zero and {@link #maxBucketsForLevel
+     *         maxBucketsForLevel(depthIndex)} minus one
      */
-    public Integer bucket(final String nodeName, final int depth) {
+    public Integer bucket(final String nodeName, final int depthIndex) {
 
-        final int byteN = hashOrder.byteN(nodeName, depth);
+        final int byteN = hashOrder.byteN(nodeName, depthIndex);
         Preconditions.checkState(byteN >= 0);
         Preconditions.checkState(byteN < 256);
 
-        final int maxBuckets = RevTree.MAX_BUCKETS;
+        final int maxBuckets = NodePathStorageOrder.maxBucketsForLevel(depthIndex);
```

Comment:
this is minor, but I believe this change means there is no need to import `org.locationtech.geogig.api.RevTree` in this class any more, correct?